Sources/MapboxMaps/Foundation/Camera/FlyToCameraAnimator.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+public class FlyToCameraAnimator: NSObject, CameraAnimator, CameraAnimatorInterface {
+
+    internal private(set) weak var delegate: CameraAnimatorDelegate?
+
+    public private(set) var owner: AnimationOwner
+
+    internal private(set) var flyToInterpolator: FlyToInterpolator?
+
+    internal private(set) var animationDuration: TimeInterval?
+
+    public private(set) var state: UIViewAnimatingState = .inactive
+
+    internal private(set) var startTime: Date?
+
+    internal private(set) var endTime: Date?
+
+    internal private(set) var finalCameraOptions: CameraOptions?
+
+    internal private(set) var animationCompletion: AnimationCompletion?
```

Comment:
Seems like many of these getters are only `internal` and not `private` for testing purposes. I think we can find some better approaches that don't force us to choose between `private` and having tests.

Reply:
refactored on my branch